### ISSUE TITLE
Allow building without internet connection when only using builtin modules

### ISSUE THREAD

**Issue by cbiesinger** [dotnet/sdk#3352]
I am in a situation where I would like to build some C# source code on a build machine that can install .deb packages but does not otherwise allow internet connections. I am only using builtin modules.

However, it seems that the dotnet SDK insists on downloading at least the linux-x64 runtime *even* when not compiling a standalone app.

Would it be possible to change the SDK to make it possible to compile non-standalone apps without needing to download anything?

**Comment by livarcocc** [comment]
What version of the .NET Core SDK are you using? We have since enabled this scenario, though it would be interesting to see your csproj. It might be possible you have something there that would trigger the need to go online.

**Comment by cbiesinger** [comment]
Package: dotnet-sdk-2.2
Version: 2.2.104-1
(from the apt-get repository)

My csproj is here:
https://github.com/MicrosoftEdge/JsDbg/blob/jsdbg_gdb_support/server/JsDbg.Core/JsDbg.Core.csproj
https://github.com/MicrosoftEdge/JsDbg/blob/jsdbg_gdb_support/server/JsDbg.Gdb/JsDbg.Gdb.csproj

I have also tried removing all the PackageReferences from the JsDbg.Core one, and dotnet/nuget still wanted to download linux-x64, even when not using --standalone / -r linux-x64.

Before I removed the PackageReferences it also wanted to download another package but I forgot which one

**Comment by livarcocc** [comment]
You have RuntimeIdentifiers set in your project here: https://github.com/MicrosoftEdge/JsDbg/blob/jsdbg_gdb_support/server/JsDbg.Gdb/JsDbg.Gdb.csproj#L6.

That tells NuGet that it should get all the assets for all the RIDs you have specified when you run restore. And, we have never supported self-contained apps (assets) in offline mode before.

If you can remove those RIDs, it should work offline for you then.

**Comment by cbiesinger** [comment]
Oh thank you, I'll try that! Clearly my unfamiliarity with dotnet is showing. Are there any downsides from removing that?

**Comment by livarcocc** [comment]
If you remove that and later on decide to publish as a self-contained app, you will hit the network then instead of during the original restore.

**Comment by livarcocc** [comment]
So, did that work for you? Ok to close this issue now?

**Comment by cbiesinger** [comment]
Yes that seems to work with a couple of minor changes (change target framework to netcoreapp2.2 and remove the LatestPatchVersion tag). Thanks!

**Comment by cbiesinger** [comment]
I haven't tried this recently but I doubt this is randomly fixed
